From 18f9f42c532dd939c4a6dde7fb1ac5d993651cd7 Mon Sep 17 00:00:00 2001
From: arun <arun.radhakrishnan@intel.com>
Date: Wed, 2 Mar 2016 00:43:37 -0600
Subject: [PATCH 18/18] iio: ltr501: fixed issue with regmap_bulk_read not
 returning correct val

it was observed that the regmap_bulk_read call within ltr501_read_als
did not update the sensor read value with every call to it,
The value was read back from the cache itself.
Updated the cache_type to REGCACHE_NONE to read the value directly from
the sensor register.

Change-Id: I9cfd0ae64e8aaf77542fb8f60d8c60f2d4c3e597
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25780
Signed-off-by: Arun Radhakrishnan <arun.radhakrishnan@intel.com>
Reviewed-on: https://android.intel.com:443/476587
Reviewed-on: https://android.intel.com:443/484462
---
 drivers/iio/light/ltr501.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/light/ltr501.c b/drivers/iio/light/ltr501.c
index 8e4002e..4d15b9e 100644
--- a/drivers/iio/light/ltr501.c
+++ b/drivers/iio/light/ltr501.c
@@ -1352,7 +1352,7 @@ static struct regmap_config ltr501_regmap_config = {
 	.reg_bits = 8,
 	.val_bits = 8,
 	.max_register = LTR501_MAX_REG,
-	.cache_type = REGCACHE_RBTREE,
+	.cache_type = REGCACHE_NONE,
 	.volatile_reg = ltr501_is_volatile_reg,
 };
 
@@ -1547,6 +1547,7 @@ static const struct acpi_device_id ltr_acpi_match[] = {
 	{"LTER0501", ltr501},
 	{"LTER0559", ltr559},
 	{"LTER0301", ltr301},
+	{"LTER0303", ltr303},
 	{"LTR303", ltr303},
 	{ },
 };
-- 
1.9.1

